ddtrace/contrib/internal/openai/utils.py

```diff
@@ -162,6 +162,136 @@ async def _extract_token_chunk(self, chunk):
             return
 
 
+class TracedOpenAIResponseStream(BaseTracedOpenAIStream):
```

Comment:
I'm seeing a lot of similarities in this traced stream class with `TracedOpenAIStream`. Is there a way we can just reuse the latter (and make some modifications to the underlying helper methods if necessary) instead of creating/maintaining a duplicate code?